### PR TITLE
feat: add tasks while cd'd into module

### LINKED ISSUE #80: feat(tasks): add task while cd'd into a module without the -c flag
Currently users must still specify the `-c` flag when adding tasks while cd'd into a module. We should change it such that a command like `add -t Task` adds a task to the current cd'd module.

### PR DESCRIPTION
**Pull request by vvidday**
closes #80 

Tried to use JCommander but decided to go with the old implementation halfway through since JCommander doesn't really have good support for conditionals AFAIK (since the same command word `add` will create either `AddCommand` or `AddTaskCommand` based on whether certain flags are present - didn't see a way to do this with JCommander). Not sure if this will be an issue @jontmy 

misc stuff (we can discuss this)
- users can still specify a module code while cd'd and it will behave as per normal

### CODE REVIEW

**Comment by domoberzin** [comment]
Actually, I wanted to suggest changing the adding module flag to something like `addm` or `mod`, so we can fully utilise JCommander. Else doing it with both tasks and modules using add doesn't seem easy. We should discuss it tomorrow, cause it seems very weird to have one parser doing half the commands and JCommander doing the rest

Also, JCommander doesnt display full error messages, unlike our legacy parser, so this might not fulfill the user story of displaying detailed error messages (UI/UX story 5)

**Comment by vvidday** [comment]
Yeah definitely agree, I went with the legacy parser mainly because I couldn't figure out how to conditionally map to different classes based on the presence/absence of a parameter in JCommander. Separating the command words will solve this issue. Sure let's discuss this tomorrow in our meeting!

**Review by domoberzin — COMMENTED** [review]
LGTM, works fine for me with some light testing, but I think we should wait for a discussion between JCommander vs legacy parser before merging this

**Comment by jontmy** [comment]
> Also, JCommander doesnt display full error messages, unlike our legacy parser, so this might not fulfill the user story of displaying detailed error messages (UI/UX story 5)

I've added implemented this in #115, but if there're any edge cases which need more detailed error messages do let me know.

File: src/main/java/modtrekt/logic/commands/AddTaskCommand.java [hunk 1]
```diff
@@ -23,14 +23,19 @@
 @Parameters(commandDescription = "Adds a task or deadline to the task book.")
 public class AddTaskCommand extends Command {
 
-    public static final String COMMAND_WORD = "add task";
-    public static final String MESSAGE_SUCCESS = "New task added: %1$s";
+    public static final String COMMAND_WORD = "add";
```
Comment (jontmy) [inline src/main/java/modtrekt/logic/commands/AddTaskCommand.java:26]: The command phrase is missing the scope - it should be `add task`.

Comment (jontmy) [inline src/main/java/modtrekt/logic/commands/AddTaskCommand.java]: The command phrase is missing the scope - it should be `add task`.

**Review by jontmy — CHANGES_REQUESTED** [review]
Great work, tested and works, just missing the scope in the command.

File: src/main/java/modtrekt/logic/commands/AddTaskCommand.java (hunk 1 above)
Comment (vvidday) [reply src/main/java/modtrekt/logic/commands/AddTaskCommand.java]: Great catch thanks!

**Review by jontmy — APPROVED** [review]
lgtm!

**Pull request merged**